src/cursor/cur_std.c

```diff
@@ -1303,15 +1275,17 @@ __wt_cursor_bound(WT_CURSOR *cursor, const char *config)
         } else
             WT_ERR_MSG(session, EINVAL,
               "a bound must be specified when setting bounds, either \"lower\" or \"upper\"");
-    } else if (have_action && WT_PREFIX_MATCH(cfg, "clear")) {
+    } else {
+        /*
+         * The config API errors out earlier if an invalid option is provided. We must be in the
+         * clear case if we're here.
+         */
```

Comment:
Probably good to add an assert for this:
```
   WT_ASSERT(session, (WT_CONF_STRING_MATCH(clear, cval)));
```

Reply:
[e45d8d3](https://github.com/wiredtiger/wiredtiger/pull/10445/commits/e45d8d33a5e839cc9b99c0415e409a13017c91cd)